### DISCUSSION TITLE
Load balancer it's only using 2 nodes instead of max

### DISCUSSION THREAD

**Discussion by fausto-spn** [istio/istio#50986] [Q&A]
Hey everyone, I'm having an issue and I couldn't find any way to fix it.
We have cluster with 10 nodes and we get multiple request load balancer is hitting only 2 nodes instead of all of them. 
I found some guys doing something similar and increasing responses [here](https://events.istio.io/istiocon-2022/slides/f2-ScalingMultiCluster.pdf)
and I tried to implement but without luck and with errors:

```shell
---
apiVersion: networking.istio.io/v1alpha3
kind: EnvoyFilter
metadata:
  name: gateway-least-request-lb-config
  namespace: istio-system
spec:
  workloadSelector:
    labels:
      istio: ingressgateway
  configPatches:
  - applyTo: NETWORK_FILTER
    match:
      context: GATEWAY
      listener:
        filterChain:
          filter:
            name: "envoy.filters.network.http_connection_manager"
    patch:
      operation: MERGE
      value:
        least_request_lb_config:
          choice_count: 10
        typed_config:
          "@type": "type.googleapis.com/envoy.extensions.filters.network.http_connection_manager.v3.HttpConnectionManager"
          server_header_transformation: PASS_THROUGH
``` 

If anyone can help me, should be great!

**Comment by abasitt** [comment]
Can you share more details.
1- Where is the requests coming from? external LB and then Istio-ingress ?
2- Which cloud environment /
3- What do you see on upstream LBs if you have? Is it discovering all the nodes endpoints ?

Last I checked all the options mentioned in that presentations has already been included with optimal defaults value in the new releases. That setting anyway will impact the requests from ingress to upstream endpoints, and not from the external clients to the ingress pods.

**Comment by fausto-spn** [comment]
Hi Abbasit, thanks for your reply!
Please check my answers below:

1- Where is the requests coming from? external LB and then Istio-ingress ?

- Internet through external LB and then to istio-ingress

2- Which cloud environment
-  AWS

3- What do you see on upstream LBs if you have? Is it discovering all the nodes endpoints ?

-  Yes, the LB is seeing all the istio-ingress GWs, but when traffic hits istio-ingress we have seen issues where traffic is only sent to two nodes/pods instead of all of them

Just to be more specific, what I'm trying to do is when we get request, we want Istio spread all of them to every available pods instead just only one or two.

Thanks in advance for your help!

**Reply by abasitt** [reply]
Thanks for sharing more details. Sorry I miss understood earlier and I thought its the requests that is coming from external LB and are not equally spread. 
Least count is the default and it should generally be spread equally.
